new/release-process.md

```diff
@@ -0,0 +1,106 @@
+---
+RFC: unassigned
+Author: Jay Mundrawala <jmundrawala@chef.io>
+Author: Bryan McLellan <btm@chef.io>
+Status: Draft
+Type: Process
+---
+
+# Chef Client Release Process
+
+This RFC describes the steps involved in releasing the Chef Client and related contingencies.
+
+## Motivation
+
+As a Chef Releaser,
+I need an established release process,
+so I can make my release consistent.
+
+As a Chef user,
+I want to know what the release process is,
+so I can install appropriate version.
+
+As a Chef user,
+I want to understand what happens when a critical bug is found,
+so I can react accordingly.
+
+## Specification
+
+### Versioning
+
+We aim to adhere to the Semantic Version scheme published [here](http://semver.org/). However, there are existing exceptions. Most notably, we do not consistently and clearly delinate which public methods are also part of the public API and guaranteed to only change with major version releases.
+
+From Semantic Versioning:
+
+```
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+MAJOR version when you make incompatible API changes,
+MINOR version when you add functionality in a backwards-compatible manner, and
+PATCH version when you make backwards-compatible bug fixes.
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
```

Comment:
Can we include something about making sure that RC versions aren't versionly-greater than the final versions? For example, I've gotten RCs in the past that I then couldn't upgrade because 10.12.1rc1 is greater than 10.12.1. We need to make sure RPM/deb/etc. will actually be able to upgrade to the final version.